### ISSUE TITLE
DINOv3ViTLayer_FP16 error

### ISSUE THREAD

**Issue by AnnaTrainingG** [huggingface/transformers#41885]
### System Info

When using the DINOv3ViTLayer in fp16 mode, the training process gets killed and throws an error.
`
exitcode : -8 (pid: 2536) 1967
error_file: <N/A> 1968
traceback : Signal 8 (SIGFPE) received by PID xxxx`

**Comment by azraattar** [comment]
@AnnaTrainingG Hi! I'd like to work on this issue. Could you please assign it to me?

I'm planning to investigate:
- Testing with fp32 instead of fp16 to isolate the precision issue
- Adding gradient clipping to prevent overflow
- Checking for NaN/Inf in the training loop
